src/protocols/secure_channel/CASESessionCache.h

```diff
@@ -0,0 +1,42 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <lib/core/CHIPError.h>
+#include <lib/core/PeerId.h>
+#include <protocols/secure_channel/CASESession.h>
+
+namespace chip {
+
+class CASESessionCache
+{
+public:
+    CASESessionCache();
+    virtual ~CASESessionCache();
+
+    CHIP_ERROR Add(CASESession & session);
+    CHIP_ERROR Remove(const ByteSpan & resumptionID);
+    CHIP_ERROR Get(const ByteSpan & resumptionID, CASESession & out_session);
+    CHIP_ERROR Get(const PeerId & peer, CASESession & out_session);
+
+private:
+    BitMapObjectPool<CASESessionSerializable, CHIP_CONFIG_CASE_SESSION_RESUME_CACHE_SIZE> mCachePool;
```

Comment:
I'm not quite sure what does this class do.

In my understanding, the session cache is an in-memory index, to index hibernated session context which is serialized into persistent storage. It is basically a map:

   (resumptionID or PeerID) -> (Storage key of `CASESessionSerializable` stored in persistent storage)

I'm not quite clear, why do we need to store `CASESessionSerializable` in a memory pool.

Reply:
I would defer to @pan-apple for this. I am new to the project and do not have know how to answer your question.

Reply:
The in memory pool can be used to cache sessions that were evicted out of Session list due to exhaustion. The size of this pool is configurable as build parameter. So if a device doesn't want to have any in RAM cache, that can established by setting the build parameter to 0.

Reply:
The potential of session resumption is to hibernate active sessions to save memory. I don't think use the in memory pool to save `CASESessionSerializable` can save memory.

I suggest save `CASESessionSerializable` to the persistent storage, and leave a key referring to it in the memory pool.